app/src/main/AndroidManifest.xml

```diff
@@ -13,6 +14,13 @@
         android:supportsRtl="true"
         android:theme="@style/AppTheme"
         android:usesCleartextTraffic="${usesCleartextTraffic}">
+        <receiver android:name=".utils.NetworkStateReceiver"
+            android:label="NetworkStateReceiver">
+            <intent-filter>
+                <action android:name="android.net.conn.CONNECTIVITY_CHANGE" />
```

Comment:
It is a deprecated method for android greater than 7.0 , please fix it 
![image](https://user-images.githubusercontent.com/48784560/116088786-d1937b80-a6bf-11eb-83e7-aa39df79c82d.png)

Reply:
Although it's deprecated but it still works fine with respect to our use case.
<img width="858" alt="Screenshot 2021-04-26 at 6 56 29 PM" src="https://user-images.githubusercontent.com/52480794/116090204-3ef3dc00-a6c1-11eb-877d-3e9c6c17f90c.png">

Reply:
@CodeChamp-SS i agree that the code is running fine but use of deprecated methods are generally discouraged, moreover  this method has limitation of working only on the main thread.

Reply:
@CodeChamp-SS i agree that code is running fine, but using deprecated methods are generally discouraged. moreover, this method is limited upto main thread only for API level 24 or higher.

Reply:
Hmm, in that case can you please suggest the best alternative to provide the required functionality?

Reply:
@CodeChamp-SS how about buiding Network State Receiver as a service ??, Service will give us more flexibility to run check API versions for selecting API specific methods and hence, eliminates the code deprecation issue in future with upcoming APIs.

Reply:
@ravi5175 thanks. @CodeChamp-SS please make the requested changes.

Reply:
> @CodeChamp-SS how about building Network State Receiver as a service ??, Service will give us more flexibility to run check API versions for selecting API specific methods and hence, eliminates the code deprecation issue in future with upcoming APIs.

hmm.., can you please give a little bit more detail on how can the NetworkStateReceiver be converted to a service? I don't know much about using services as of now 😅